### PR TITLE
[website] Automate website artifacts uploading

### PR DESCRIPTION
**Pull request by waytrue17**
## Description ##
Port #19244 as a part of the MXNet release process for 2.0+ versions.  Also added a back-up mechanism to the versions folder. The back-up artifact will have name `versions_backup_upto_$(release_version).zip'

### Essentials ###
- [ ] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by waytrue17** [comment]
@ys2843 @leezu Can you review?

File: ci/jenkins/Jenkins_steps.groovy [hunk 1]
```diff
@@ -1168,6 +1192,28 @@ def docs_publish_beta() {
...
 
+// This is for uploading website artifacts to S3 bucket
+// Assumes you have run docs_full_website function
+def docs_upload_s3() {
+    return ['Upload artifacts to s3 bucket': {
+      node(NODE_LINUX_CPU) {
+        ws('workspace/docs') {
+          timeout(time: max_time, unit: 'MINUTES') {
+            if(env.FOLDER_NAME) {
+              utils.unpack_and_init('full_website', 'docs/_build/full_website.tgz')
```
Comment (szha) [inline ci/jenkins/Jenkins_steps.groovy:1203]: can we make sure we have backups and can roll back if there's any issue.

Comment (waytrue17) [reply ci/jenkins/Jenkins_steps.groovy:1203]: Added a back-up mechanism @szha

File: ci/docker/runtime_functions.sh [hunk 2]
```diff
@@ -1212,6 +1212,30 @@ build_docs_beta() {
...
+        exit 1
+    fi
+    mv master $folder_name
+    popd
+    # back up versions folder
+    wget https://mxnet-website-static-artifacts.s3.us-east-2.amazonaws.com/versions.zip -O versions-backup.zip
+    aws s3 cp versions-backup.zip s3://mxnet-website-static-artifacts --acl public-read
+
+    zip -r9 versions.zip versions/.
+    aws s3 cp versions.zip s3://mxnet-website-static-artifacts --acl public-read
```
Comment (szha) [inline ci/docker/runtime_functions.sh:1235]: one version might not be enough given that the pipeline runs daily and issues can go unnoticed longer.

Comment (marcoabreu) [reply ci/docker/runtime_functions.sh:1235]: How about just publishing them in the format "YEAR-MONTH-DAY-COMMITHASH.zip" with a policy to recycle them after 3 months without access?

Comment (waytrue17) [reply ci/docker/runtime_functions.sh]: This pipeline is for MXNet releases and should only be triggered to update the release version of website artifact. I think backing up by versions should be enough in this case. I changed the code to archive each released versions so the new back-up won't override the previous ones. The back-ups will have name `versions_backup_upto_$(release_version).zip'

**Comment by waytrue17** [comment]
@mxnet-bot run ci [centos-cpu, centos-gpu]

**Comment by waytrue17** [comment]
@mxnet-bot run ci [unix-cpu]

File: ci/docker/runtime_functions.sh [hunk 3]
```diff
@@ -1212,6 +1212,30 @@ build_docs_beta() {
...
+        exit 1
+    fi
+    mv master $folder_name
+    popd
+    zip -r9 versions.zip versions/.
+    # Upload versions folder
+    aws s3 cp versions.zip s3://mxnet-website-static-artifacts --acl public-read
+    # Backup versions folder with the latest version name
+    backup_file="versions_backup_upto_$folder_name.zip"
+    aws s3 cp versions.zip s3://mxnet-website-static-artifacts/$backup_file --acl public-read
```
Comment (szha) [inline ci/docker/runtime_functions.sh:1235]: Copy between s3 locations will make it more efficient.

**Review by szha — APPROVED** [review]

**Comment by szha** [comment]
Seems that website builds started failing due to permission issue:
https://jenkins.mxnet-ci.amazon-ml.com/blue/organizations/jenkins/mxnet-validation%2Fwebsite/detail/PR-19986/1/pipeline

**Comment by waytrue17** [comment]
> Seems that website builds started failing due to permission issue:
> https://jenkins.mxnet-ci.amazon-ml.com/blue/organizations/jenkins/mxnet-validation%2Fwebsite/detail/PR-19986/1/pipeline

That's due to an s3 object permission setup issue when I manually uploading the versions.zip for testing. It's been fixed. This PR shouldn't have the issue as we set `--acl public-read` when uploading the file.

**Review by Zha0q1 — APPROVED** [review]
LGTM

**Pull request merged**